prow/config/config.go

```diff
@@ -216,14 +216,28 @@ type InRepoConfig struct {
 	AllowedClusters map[string][]string `json:"allowed_clusters,omitempty"`
 }
 
+func SplitRepoName(repo string) (string, string, error) {
+	// Normalize repo name to remove http:// or https://, this is the case for some
+	// of the gerrit instances.
+	repo = strings.TrimPrefix(repo, "http://")
+	repo = strings.TrimPrefix(repo, "https://")
+	s := strings.SplitN(repo, "/", 2)
+	if len(s) != 2 {
+		return "", "", fmt.Errorf("repo %s cannot be split into org/repo", repo)
+	}
+	return s[0], s[1], nil
+}
+
 // InRepoConfigEnabled returns whether InRepoConfig is enabled for a given repository.
 func (c *Config) InRepoConfigEnabled(identifier string) bool {
+
+	org, repo, err := SplitRepoName(identifier)
 	if c.InRepoConfig.Enabled[identifier] != nil {
-		return *c.InRepoConfig.Enabled[identifier]
+		return *c.InRepoConfig.Enabled[strings.Join([]string{org, repo}, "/")]
 	}
-	identifierSlashSplit := strings.Split(identifier, "/")
-	if len(identifierSlashSplit) == 2 && c.InRepoConfig.Enabled[identifierSlashSplit[0]] != nil {
-		return *c.InRepoConfig.Enabled[identifierSlashSplit[0]]
+	// If err then it cannot be split.
+	if c.InRepoConfig.Enabled[org] != nil && err == nil {
```

Comment:
If `err != nil` then both org and repo are empty string, then the check above would evaluate `/` instead, which doesn't seem safe. Should this error check happen earlier?

Reply:
fixed